### DISCUSSION TITLE
Clarification on Execution Order of entry Actions and invoke

### DISCUSSION THREAD

**Discussion by ninjz** [statelyai/xstate#4948] [Q&A]
Are there any guarantees that actions specified in the `entry` property are completed before the `invoke` property is called? Specifically, I need to ensure that the context is properly set up using `assign` within the `entry` property before invoking an actor via `invoke`. Can you provide clarity on this behavior?

**Comment by Andarist** [comment]
IIRC, it's guaranteed. We have tests for this behavior
